app/code/Magento/Wishlist/view/frontend/templates/sidebar.phtml

```diff
@@ -9,7 +9,7 @@
 <?php
 $wishlistHelper = $this->helper(\Magento\Wishlist\Helper\Data::class);
 ?>
-<?php if ($wishlistHelper->isAllow()) : ?>
+<?php if ($wishlistHelper->isAllow()  && $wishlistHelper->ifNotWishlistView()) : ?>
```

Comment:
What about solving this through the layout? `Magento/Wishlist/view/frontend/layout/default.xml` adds the sidebar so it can also be removed in `Magento/Wishlist/view/frontend/layout/wishlist_index_index.xml` rather than conditionally outputting HTML in the template.

Reply:
Hi, 
someone tried that way here https://github.com/magento/magento2/pull/25345/files and magento testing failed that pr, hence I tried this way.

and yes, it is possible to remove using layout too.

Reply:
Is there anything to be done on this?